R/cut.R

```diff
@@ -0,0 +1,140 @@
+#' The leftmost interval corresponds to level one, the next leftmost to level two and 
+#' so on.
+#' @param x A crunch variable of the class NumericVariable
+#' @param breaks Either a numeric vector of two or more unique cut points 
+#' or a single number (greater than or equal to 2) giving the number of intervals 
+#' into which x is to be cut.
+#' @param variable.name The name of the resulting case variable as a character string. 
+#' @param labels labels for the levels of the resulting category.
+#' By default, labels are constructed using interval notation.
+#' If labels = FALSE, simple integer codes are returned instead of a factor.
+#' @param include.lowest logical, indicating if an `x[i]` equal to the lowest 
+#' (or highest, for right = FALSE) `breaks` value should be included.
+#' @param right logical, indicating if the intervals should be closed on the right 
+#' (and open on the left) or vice versa.
+#' @param dig.lab	integer which is used when labels are not given. 
+#' It determines the number of digits used in formatting the break numbers.
+#' @param ordered_result	Ignored.
+#' @param ... further arguments passed to or from other methods.
+#'
+#'
+#' @return a Crunch VariableDefinition
+#' @export
+#'
+#' @examples
+#' \dontrun{
+#' ds <- loadDataset("mtcars")
+#' ds$cat_var <- cut(ds$mpg, 3, variableName = "new_var")
+#' }
+#'  
+setMethod("cut", "NumericVariable", function(x, 
+                                             breaks, 
+                                             variable.name, 
+                                             labels = NULL,
+                                             include.lowest = FALSE,
+                                             right = TRUE,
+                                             dig.lab = 3,
+                                             ordered_result = FALSE, ...){
+    env <- environment()
+      if (missing(variable.name)) {
+        halt("Must provide the name for the new variable")
+    }
+    if (length(breaks) == 1L) {
+        if (is.na(breaks) || breaks < 2L) {
+            halt("invalid number of intervals")
+        }
+        nb <- as.integer(breaks + 1) # one more than #{intervals}
+        rx <- c(min(x, na.rm = TRUE), max(x, na.rm = TRUE))
+        dx <- diff(rx)
+        if (dx == 0) {
+            dx <- abs(rx[1L])
+            breaks <- seq.int(rx[1L] - dx/1000, rx[2L] + dx/1000,
+                              length.out = nb)
+        } else {
+            breaks <- seq.int(rx[1L], rx[2L], length.out = nb)
+            breaks[c(1L, nb)] <- c(rx[1L] - dx/1000, rx[2L] + dx/1000)
+        }
+    }else {
+        breaks <- sort.int(as.double(breaks))
+        nb <- length(breaks)
+    }
+    if (anyDuplicated(breaks)) halt("'breaks' are not unique")
+    if (is.null(labels)) { #Autogenerate labels if not supplied
+      labels <- crunch:::generateCutLabels(dig.lab, breaks, nb, right, include.lowest)
```

Comment:
don't need to ::: a function in your own package. `check` will whine about this, I think.